services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,42 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  // If no event is given arguments are bad
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+  // If queue handle doesnt describe a queue, error in state
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+  // If queue is full, cannot add event, error
+  if (xQueueSend(thermalMgrQueueHandle, event, 10) == errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t interrupt = {THERMAL_MGR_INTERRUPT_EVENT};
```

Comment:
C99 allows you to set elements of a struct by name when initializing. So it's more readable to do something like: `thermal_mgr_event_t interrupt = { .type=THERMAL_MGR_INTERRUPT_EVENT }`.